pkg/nixpath/nixpath.go

```diff
@@ -18,9 +18,8 @@ const (
 var (
 	NameRe = regexp.MustCompile(`[a-zA-Z0-9+\-_?=][.a-zA-Z0-9+\-_?=]*`)
 	PathRe = regexp.MustCompile(fmt.Sprintf(
-		`^%v/([%v]{%d})-(%v)$`,
+		`^%v/([0123456789abcdfghijklmnpqrsvwxyz]{%d})-(%v)$`,
 		regexp.QuoteMeta(StoreDir),
-		nixbase32.Alphabet,
```

Comment:
See me above comment about why I left this exported.

Reply:
Can you help me understand why this regexp is exported? It's not used in this package, nor is it used by any importers AFAICT.

In my experience, exporting a regexp isn't a very flexible API. If it's being used for validation, then it prevents using higher-level validation later on.

Reply:
nixpath _is_ exporting a regex as a symbol, but I understand that `nixbase32.Alphabet` is just exporting a string, which is more reasonable.

That said, for something like the router usage you link, I would expect users to match on a `.narinfo` suffix and then check for existence on objects. `base32.ValidateString` can be used as a quick check that's more precise than using the alphabet for a regex. That's why I'm leaning toward unexporting the `Alphabet` constant: there are better ways for users of the library to accomplish the same goals.

Reply:
I think I'd prefer keeping nixbase32.Alphabet exported for such usecases.

Reply:
Done.